test/system/src/main/java/io/pravega/test/system/framework/services/docker/PravegaSegmentStoreDockerService.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.test.system.framework.services.docker;
+
+import com.google.common.base.Strings;
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.mount.Mount;
+import com.spotify.docker.client.messages.swarm.ContainerSpec;
+import com.spotify.docker.client.messages.swarm.EndpointSpec;
+import com.spotify.docker.client.messages.swarm.NetworkAttachmentConfig;
+import com.spotify.docker.client.messages.swarm.PortConfig;
+import com.spotify.docker.client.messages.swarm.ResourceRequirements;
+import com.spotify.docker.client.messages.swarm.Resources;
+import com.spotify.docker.client.messages.swarm.ServiceMode;
+import com.spotify.docker.client.messages.swarm.ServiceSpec;
+import com.spotify.docker.client.messages.swarm.TaskSpec;
+import java.net.URI;
+import lombok.extern.slf4j.Slf4j;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import static io.pravega.test.system.framework.Utils.DOCKER_NETWORK;
+
+@Slf4j
+public class PravegaSegmentStoreDockerService extends DockerBasedService {
+
+    private static final int SEGMENTSTORE_PORT = 12345;
```

Comment:
Is making the port constant going to cause conflicts? Should we randomize port instead?

Reply:
No, it shouldnt cause any problem and also the docker file used to build segmentstore image exposes this ports. The docker client needs to access only the port `2375` of master node. Apart from that all the other ports on master and slave nodes are free.
zk : 2181
bk: 3181
controller :9090, 9091
segmentstore: 12345.

So, it won't be a problem.

Reply:
Ok.